crates/pep508-rs/src/marker/algebra.rs

```diff
@@ -748,6 +751,110 @@ impl Edges {
     }
 }
 
+// Normalize a [`VersionSpecifier`] before adding it to the tree.
+fn normalize_specifier(specifier: VersionSpecifier) -> VersionSpecifier {
+    let (operator, version) = specifier.into_parts();
+
+    // The decision diagram relies on the assumption that the negation of a marker tree is
+    // the complement of the marker space. However, pre-release versions violate this assumption.
+    // For example, the marker `python_full_version > '3.9' or python_full_version <= '3.9'`
+    // does not match `python_full_version == 3.9.0a0`. However, it's negation,
+    // `python_full_version > '3.9' and python_full_version <= '3.9'` also does not include
+    // `3.9.0a0`, and is actually `false`.
```

Comment:
```suggestion
    // `3.9.0a0`, and always evaluates to `false`.
```
Is this what you're saying?